sysconfig/cloudinit.go

```diff
@@ -76,21 +80,50 @@ func installCloudInitCfg(src, targetdir string) error {
 	return nil
 }
 
-// TODO:UC20: - allow cloud.conf coming from the gadget
-//            - think about if/what cloud-init means on "secured" models
+// installUnifiedCloudInitCfg installs a single cloud-init config file to the
+// cloud config dir as "ubuntu-core-cloud_91.cfg".
+func installUnifiedCloudInitCfg(src, targetdir string) error {
+	ubuntuDataCloudCfgDir := filepath.Join(ubuntuDataCloudDir(targetdir), "cloud.cfg.d/")
+	if err := os.MkdirAll(ubuntuDataCloudCfgDir, 0755); err != nil {
+		return fmt.Errorf("cannot make cloud config dir: %v", err)
+	}
+
+	configFile := filepath.Join(ubuntuDataCloudCfgDir, "80_device_gadget.cfg")
+	return osutil.CopyFile(src, configFile, 0)
+}
+
 func configureCloudInit(opts *Options) (err error) {
 	if opts.TargetRootDir == "" {
 		return fmt.Errorf("unable to configure cloud-init, missing target dir")
 	}
 
-	switch opts.CloudInitSrcDir {
-	case "":
-		// disable cloud-init by default using the writable dir
-		err = DisableCloudInit(WritableDefaultsDir(opts.TargetRootDir))
-	default:
-		err = installCloudInitCfg(opts.CloudInitSrcDir, WritableDefaultsDir(opts.TargetRootDir))
+	// TODO: too naive to just re-use GadgetDir here automatically?
```

Comment:
naive in which sense?

Reply:
naive in that maybe devicestate should pass another option to sysconfig telling it to explicitly use and look for the gadget cloud.conf, rather than sysconfig just implicitly copying it if it exists

in other words, should the decision to copy cloud.conf be made in sysconfig or the caller (i.e. devicestate)

Reply:
ah, I see we can drop the comment,  the caller of this has the comment:

// ConfigureRunSystem configures the ubuntu-data partition with any
// configuration needed from e.g. the gadget or for cloud-init.

maybe that just needs to be expand with "...including gadget configuration for cloud-init"

Reply:
I updated the comment on ConfigureRunSystem as well